### PR TITLE
AOP2-35 Admin page  - update ceiling price

### PR DESCRIPTION
**Pull request by bsigney**

### CODE REVIEW

File: backend/app/api/services/ceiling_prices.py [hunk 1]
```diff
@@ -76,9 +76,11 @@ def update_ceiling_price(self, ceiling_id, new_ceiling, set_current_price_to_cei
             ceiling_price.supplier_code,
             ceiling_price.service_type_id,
             ceiling_price.sub_service_id,
-            pendulum.today(current_app.config['DEADLINES_TZ_NAME']).date())
+            pendulum.today(current_app.config['DEADLINES_TZ_NAME']).date(),
+            ceiling_price.id)
```
Comment (qimyi) [inline backend/app/api/services/ceiling_prices.py]: we are passing 4 properties of ceiling_price to get_prices(). Is it possible to just pass ceiling_price instead and access the props within the method itself?

File: backend/app/api/services/prices.py [hunk 2]
```diff
@@ -12,18 +12,26 @@ def __init__(self, *args, **kwargs):
...
-                    ServiceTypePrice.is_current_price(date))\
-            .distinct(Region.state, Region.name, ServiceTypePrice.supplier_code, ServiceTypePrice.service_type_id,
-                      ServiceTypePrice.sub_service_id, ServiceTypePrice.region_id)\
-            .order_by(Region.state, Region.name, ServiceTypePrice.supplier_code.desc(),
-                      ServiceTypePrice.service_type_id.desc(), ServiceTypePrice.sub_service_id.desc(),
-                      ServiceTypePrice.region_id.desc(), ServiceTypePrice.updated_at.desc())\
+                    ServiceTypePrice.is_current_price(date))
+
+        if ceiling_price_id:
+            prices = prices.filter(
```
Comment (waleedusman) [inline backend/app/api/services/prices.py]: Why do we filter by ceiling_id? "Current price with ceiling_id X" is not current price.

File: frontend/apps/orams/actions/editPriceActions.js [hunk 3]
```diff
@@ -124,7 +139,7 @@ export function setUserPrice(price, capPrice) {
   return (dispatch, getState) => {
     const state = getState()
     const priceObj = {
-      capPrice: capPrice,
+      capPrice,
```
Comment (qimyi) [inline frontend/apps/orams/actions/editPriceActions.js]: if possible, rename this to ceiling price for consistency with b/e

File: backend/app/api/services/ceiling_prices.py [hunk 4]
```diff
@@ -76,9 +76,11 @@ def update_ceiling_price(self, ceiling_id, new_ceiling, set_current_price_to_cei
...
         if supplier_prices:
             current_price = supplier_prices[0]['price']
+            current_price = current_price.translate(None, ',')
```
Comment (waleedusman) [inline backend/app/api/services/ceiling_prices.py]: what does `translate()` do?

File: frontend/apps/orams/components/EditCeilingPriceForm/EditCeilingPriceForm.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        </div>
+      </header>
+    )
+  }
+
+  render() {
+    const { model, action, submitClicked, priceData, handlePriceSubmit } = this.props
+    let hasFocused = false
+
+    const setFocus = e => {
```
Comment (qimyi) [inline frontend/apps/orams/components/EditCeilingPriceForm/EditCeilingPriceForm.js]: I've never seen setting focus like this
👍 
TIL

File: frontend/apps/orams/components/PricingList/PricingList.js [hunk 6]
```diff
@@ -1,22 +1,63 @@
-/* eslint-disable */
 import React from 'react'
+import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
```
Comment (qimyi) [inline frontend/apps/orams/components/PricingList/PricingList.js]: Would prefer this to be just called `Alert`, what do you think?

File: frontend/apps/orams/pages/AdminPage.js [hunk 7]
```diff
@@ -103,9 +107,10 @@ const mapStateToProps = state => {
   }
 }
 
-const mapDispatchToProps = {
-  doSearchSupplier: (searchTerm) => searchSupplier(searchTerm),
-  doSearchUser: (searchTerm) => searchUser(searchTerm)
-}
+const mapDispatchToProps = dispatch => ({
+  doSearchSupplier: searchTerm => dispatch(searchSupplier(searchTerm)),
+  doSearchUser: searchTerm => dispatch(searchUser(searchTerm)),
```
Comment (qimyi) [inline frontend/apps/orams/pages/AdminPage.js]: I think you don't need to dispatch on line 111 and 112, please double check

File: backend/app/api/services/prices.py (hunk 2 above)
Comment (bsigney) [reply backend/app/api/services/prices.py]: Without this check, this was returning ALL current prices for Supplier, Service type, and sub service.  Which means that it is not returning current price for a particular ceiling_price_id.

End result was, when the `setCurrentPriceToCeiling` was true, the wrong price was getting updated.

File: backend/app/api/services/ceiling_prices.py (hunk 4 above)
Comment (bsigney) [reply backend/app/api/services/ceiling_prices.py]: removes the `,` character.
The serialized version of the price has `,` character in it.

File: backend/app/api/services/prices.py [hunk 8]
```diff
@@ -12,18 +12,26 @@ def __init__(self, *args, **kwargs):
         super(PricesService, self).__init__(*args, **kwargs)
         self.audit = AuditService()
 
-    def get_prices(self, code, service_type_id, category_id, date):
+    def get_prices(self, code, service_type_id, category_id, date, region_id=None):
```
Comment (waleedusman) [inline backend/app/api/services/prices.py]: Ideally you would add some test(s) to verify the behaviour you want, i.e. if a service has current prices for multiple regions you want to be able to retrieve the current price for a specific region.

File: backend/tests/fixtures/test_ceiling_prices.py [hunk 9]
```diff
@@ -256,3 +256,39 @@ def test_current_price_created_when_all_expired(
     assert len(prices) == 2
     assert prices[0].date_from == pendulum.today().date()
     assert prices[0].price == 310.00
+
+
+def test_current_price_for_a_region_updated_when_matching_required(
```
Comment (waleedusman) [inline backend/tests/fixtures/test_ceiling_prices.py]: 👍 💯

**Pull request merged**